balboa-service-kafka/src/main/scala/com/socrata/balboa/service/kafka/consumer/PersistentKafkaConsumerComponent.scala

```diff
@@ -45,9 +44,15 @@ trait PersistentKafkaConsumerComponent[K,M] extends KafkaConsumerComponent[K,M]
      * See [[ConsumerLike.consume()]].
      */
     @annotation.tailrec
-    override final def consume(key: K, message: M): Option[String] = {
+    override final def consume(key: K, message: M, attempts: Int = 0): Option[String] = {
       // Note: Scala compiler will unravel this tail recursion into a while loop,
       // As an effect, Having the node or killing the service will potentially result in a message loss
+      
+      if(attempts >= retries) {
+        Log.error("Exceeded allowed number of retries")
+        throw new IOException("Exceeded allowed number of retries")
```

Comment:
You think a more descriptive exception would be helpful if this would ever occur?